datafusion/common/src/scalar/mod.rs

```diff
@@ -6331,7 +6331,7 @@ mod tests {
 
     #[test]
     // despite clippy claiming they are useless, the code doesn't compile otherwise.
-    #[allow(clippy::useless_vec)]
+    #[expect(clippy::useless_vec)]
```

Comment:
Have we tried removing this lint and fixing the clippy lint as it occurs to see if we actually need this expect?